parlai/core/logging_utils.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import sys
+import logging
+from pathlib import Path
+from datetime import date
+
+INFO_LEVEL = logging.INFO
+DEBUG_LEVEL = logging.DEBUG
+WARN_LEVEL = logging.WARNING
+ERROR_LEVEL = logging.ERROR
+CRITICAL_LEVEL = logging.CRITICAL
+
+
+class ParlaiLogger(logging.Logger):
+    def __init__(
+        self,
+        name,
+        console_level,
+        console_format=None,
+        file_format=None,
+        file_level=logging.INFO,
+        filename=None,
+    ):
+        """
+        Initialize the logger object
+        :param name: Name of the logger
+        :param console_level: min. Level of messages logged to console
+        :param console_format: The format of messages logged to the console. Simple stdout is used if None specified
+        :param file_format: The format of messages logged to the file (A default is used if None specified)
+        :param file_level: min. Level of messages logged to the file
+        :param filename: The file the logs are written to
+
+        A default file is initialized if no filename was provided. It can be
+        found in ParlAI/Logs/*year*/*month*/*YYYY-MM-DD*.log
+        """
+        # Setup logging location if none specified
+        if filename is None:
+            today = date.today().strftime("%d/%m/%Y").split("/")
+            day, month, year = today[0], today[1], today[2]
+            proj_root = str(Path(__file__).parent.parent.parent)
+            log_dir_path = proj_root + "/" + "Logs/" + year + "/" + month
+            log_fname = year + "-" + month + "-" + day + ".log"
+            if not os.path.exists(log_dir_path):
+                os.makedirs(log_dir_path)
+            filename = os.path.join(proj_root, log_dir_path, log_fname)
+        # Default format used if no file format provided
+        if file_format is None:
+            file_format = '%(asctime)s :: %(levelname)s :: %(message)s'
```

Comment:
We might be able to amp this up, no?

Reply:
https://github.com/facebookresearch/ParlAI/pull/2030#discussion_r330243127

Reply:
Seems like Kurt and I disagree. He gets to make the call.